### PR TITLE
add store selector

### LINKED ISSUE #958: Store switcher
Would be helpful if there was a quick way to switch stores - particularly when working with requisitions and testing across different stores.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #958

While not strictly speaking a requirement.. it is handy. 
Just popped in a sneaky `onClick` on the store, pop up a list of stores, click and change. No one needs to know..

Screen Shot 2022-03-04 at 4 39 28 PM

### CODE REVIEW

File: packages/common/src/authentication/AuthContext.tsx [hunk 1]
```diff
@@ -107,6 +109,13 @@ export const AuthProvider: FC = ({ children }) => {
     return { token, error };
   };
 
+  const setStore = (store: Store) => {
+    if (!localToken) return;
+
+    setLocalStore(store);
+    setMRUCredentials({ username: user?.name ?? '', store: store });
```
Comment (mark-prins) [inline packages/common/src/authentication/AuthContext.tsx:116]: undecided whether this is a good idea. Went with it anyway.. 🤔

File: packages/host/src/components/Footer/StoreSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+} from '@openmsupply-client/common';
+import { StoreRowFragment, useStores } from '@openmsupply-client/system';
+
+export const StoreSelector: FC = ({ children }) => {
+  const { store, setStore } = useAuthContext();
+  const { hide, PaperClickPopover } = usePaperClickPopover();
+  const { data, isLoading } = useStores();
+  const t = useTranslation('app');
+
+  const storeSorter = (a: StoreRowFragment, b: StoreRowFragment) => {
```
Comment (mark-prins) [inline packages/host/src/components/Footer/StoreSelector.tsx:19]: yes, duplicated from login. I still hope that the api will provide a sort option

File: packages/host/src/components/Footer/StoreSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <PaperClickPopover
+      placement="top"
+      width={250}
+      Content={
+        <PaperPopoverSection label={t('select-store')}>
+          {isLoading ? (
+            <CircularProgress size={12} />
+          ) : (
+            <Box
+              style={{ maxHeight: '200px', overflowY: 'auto' }}
```
Comment (mark-prins) [inline packages/host/src/components/Footer/StoreSelector.tsx:51]: limiting height just in case there's a lot of them 🤷

Comment (joshxg) [reply packages/host/src/components/Footer/StoreSelector.tsx:51]: good idea

**Review by joshxg — APPROVED** [review]
Wow the efficiency! the productivity! The switchiness! 

My life has changed

**Comment by joshxg** [comment]
Cheeky commit for the undefined return (and a compiler error in the footer) for the cheeky store selector. Woo I'm excited

**Pull request merged**